processing/src/main/java/org/apache/druid/segment/nested/GlobalDimensionDictionary.java

```diff
@@ -42,45 +47,107 @@
   private final ComparatorDimensionDictionary<String> stringDictionary;
   private final ComparatorDimensionDictionary<Long> longDictionary;
   private final ComparatorDimensionDictionary<Double> doubleDictionary;
+  private final Set<Object[]> stringArrays;
+  private final Set<Object[]> longArrays;
+  private final Set<Object[]> doubleArrays;
```

Comment:
Why this separation by type?  Isn't is just an `array` and the ids are any other scalar thing in the global dictionary?

Reply:
this is also used for merging segments, so the original ints they referred to might have been remapped when the string/long/double dictionaries are merged beneath them, so we have to round trip like this to lookup the new values

Reply:
oh maybe this isn't used for merging segments... let me get back to you on this one